### DISCUSSION TITLE
405 Method Not Allowed after deployment on vercel

### DISCUSSION THREAD

**Discussion by NicolasChusseau** [vercel/community#2470] [Help]
### Summary

Hi everyone ! 

 I have an app using redwood, supabase and prisma.

I've just deployed it on vercel and now I can't access to my data. I have an error 405 Method Not Allowed, but I don't understand why, as it works well on localhost.
I didn't forget to add env var, but I don't have vercel.json file as I don't know how to write it.

### Additional information

Actual : 
![image](https://github.com/vercel/community/assets/122353423/d1b8ec30-1115-47a1-8148-49a4272cccbd)
![image](https://github.com/vercel/community/assets/122353423/a8332678-08de-4a60-8d8a-a3988facd32a)

Expected (it works on localhost) :
![image](https://github.com/vercel/community/assets/122353423/1cf487cb-f8ff-4694-bdcd-c5bea08ebd3b)

**Comment by amyegan** [comment]
A 405 [INVALID_REQUEST_METHOD](https://vercel.com/docs/errors/application-error-codes#invalid_request_method) error is thrown when a static resource is requested with a type other than GET or HEAD. A serverless environment works a little differently from a more traditional server setup, so code needs to be adjusted to run in serverless or edge functions. Here are a few resources to help you get it working:
- [How to Deploy a Redwood Site with Vercel](https://vercel.com/guides/deploying-redwood-with-vercel)
- [Using an ORM to access your Postgres database: Prisma](https://vercel.com/docs/storage/vercel-postgres/using-an-orm#prisma)
- [Vercel Supabase integration](https://vercel.com/integrations/supabase)
- [Vercel Prisma starter templates](https://vercel.com/templates?search=prisma)

**Reply by NicolasChusseau** [reply]
I have tried to follow the instructions in these various links but I still can't get rid of my 405 error.
I don't understand what I'm doing wrong.
Can anyone help me ?
Here my schema.prisma : 
```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
  directUrl = env("DIRECT_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = "native"
  previewFeatures = ["jsonProtocol"]
}

model post {
  id        Int      @id @default(autoincrement())
  title     String
  body      String
  createdAt DateTime @default(now())
}

model contact {
  id        Int      @id @default(autoincrement())
  name      String
  email     String
  message   String
  createdAt DateTime @default(now())
}

model user {
  id                  Int       @id @default(autoincrement())
  name                String?
  email               String    @unique
  hashedPassword      String
  salt                String
  resetToken          String?
  resetTokenExpiresAt DateTime?
}
```

My .env file : 
```
SESSION_SECRET=something
DATABASE_URL=postgresql://postgres:password@db.fcmaslvbiylgjuqudkew.supabase.co:5432/postgres?pgbouncer=true
DIRECT_URL=postgresql://postgres:password@db.fcmaslvbiylgjuqudkew.supabase.co:5432/postgres
```

If you need more information, please tell me.

**Comment by recallwei** [comment]
Try to add the following code in your `vercel.json`, the destination should be replaced with your API endpoint.

```json
{
  "rewrites": [
    {
      "source": "/api/:match*",
      "destination": "http://localhost/:match*"
    },
  ]
}
```

**Reply by NicolasChusseau** [reply]
I've added it to my `vercel.json`. I've changed the destination by something like this : `"destination": "https://somthing.vercel.app/.redwood/functions/graphql/:match*` but I still get error 405.

**Comment by cbess** [comment]


**Reply by cbess** [reply]
I noticed, in my testing, if you have an error/exception in your handler, it can cause your API in the preview deployment to trigger a `405` error 🤷🏾

**Comment by cbess** [comment]
@NicolasChusseau check your Logs for the deployment that is giving you the `405`, that may indicate the reason for the returned status code

**Comment by Alberto-Cortes** [comment]
Hey, did this got solved? Facing the same issue right now

**Reply by NicolasChusseau** [reply]
I stopped working on it in July 2023. To make it work, I launched my application on render instead. So I haven't found a real solution.
